src/main/scala/org/apache/spark/scheduler/cluster/armada/ArmadaClusterManagerBackend.scala

```diff
@@ -43,15 +38,12 @@ private[spark] class ArmadaClusterManagerBackend(
     masterURL: String)
   extends CoarseGrainedSchedulerBackend(scheduler, sc.env.rpcEnv) {
 
-  // FIXME
-  private val appId = "fake_app_id_FIXME"
 
   private val initialExecutors = SchedulerBackendUtils.getInitialTargetExecutorNumber(conf)
   private val executorTracker = new ExecutorTracker(new SystemClock(), initialExecutors)
-  private val gangAnnotations = GangSchedulingAnnotations(None, initialExecutors, conf.get(GANG_SCHEDULING_NODE_UNIFORMITY_LABEL))
 
   override def applicationId(): String = {
-    conf.getOption("spark.app.id").getOrElse(appId)
+    conf.getOption("spark.app.id").getOrElse(DEFAULT_ARMADA_APP_ID)
```

Comment:
At this point in time, `spark.app.id` should exist, so this could be simplified to
```suggestion
    conf.getAppId
```